src/cargo/core/compiler/standard_lib.rs

```diff
@@ -78,6 +78,7 @@ pub fn resolve_std<'gctx>(
     // TODO: Consider doing something to enforce --locked? Or to prevent the
     // lock file from being written, such as setting ephemeral.
     let mut std_ws = Workspace::new(&std_ws_manifest_path, gctx)?;
+    std_ws.set_is_std(true);
```

Comment:
Should we name this for the specific policy involved (forcing the workspace to be locked, independent of any flags)?